### PR TITLE
1833 - made the language list show the native language

### LINKED ISSUE #1833: Make language list show the native names for the languages
So instead of just the language code, there would be the language name after it (this can probably be fetched from C#, with caching), so `fi` would be `fi (Suomi)` after this.
Because this list looks pretty confusing:

### PR DESCRIPTION
**Pull request by zackdrake**
language list now show the native names for the languages in parenthesis

closes #1833

### CODE REVIEW

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]
This doesn't look correct to me:

Instead of showing what the native name of English is in English each language should show the native name of *that* language.

**Comment by zackdrake** [comment]
My apologies, it's fixed now.
I had to change a function in settings from private to public and change it's place in the code for the CI. However i'm not sure if it's the right way to do it. If there's a better way or if you would like me to change something please tell me and i will oblige

**Comment by hhyyrylainen** [comment]
That's the right way to make it public.

**Comment by Dragonmemo** [comment]
I was in French, then clicked on the english language, the game crashed with this error on the logs :

ERROR: Unhandled Exception: System.Globalization.CultureNotFoundException: Culture name en(English) is not supported. Parameter name: name at System.Globalization.CultureInfo..ctor (System.String name, System.Boolean useUserOverride, System.Boolean read_only) [0x00073] in <34e7ab706f1141f9b981d74fbe2f5560>:0 at System.Globalization.CultureInfo..ctor (System.String name, System.Boolean useUserOverride) [0x00000] in <34e7ab706f1141f9b981d74fbe2f5560>:0 at System.Globalization.CultureInfo..ctor (System.String name) [0x00000] in <34e7ab706f1141f9b981d74fbe2f5560>:0 at Settings.GetCultureInfo (System.String language) [0x00017] in <20165d82c7cd4d8e9e0ec1929de097ea>:0 at Settings.ApplyLanguageSettings () [0x00022] in <20165d82c7cd4d8e9e0ec1929de097ea>:0 at OptionsMenu.OnLanguageSettingSelected (System.Int32 item) [0x0002c] in <20165d82c7cd4d8e9e0ec1929de097ea>:0
Godot Engine v3.2.3.stable.mono.official - https://godotengine.org
OpenGL ES 3.0 Renderer: Radeon(TM) RX 460 Graphics

Else everything seems to be shown perfectly, unless they are stuff not yet translated.

**Comment by zackdrake** [comment]
I used the index to can look up the simple name in the languages list as instructed instead of reverting to the original name

File: src/engine/Settings.cs [hunk 1]
```diff
@@ -349,7 +385,6 @@ public void ApplyAll(bool delayedApply = false)
             ApplySoundSettings();
         }
 
-        ApplyLanguageSettings();
```
Comment (FuJa0815) [inline src/engine/Settings.cs:352]: Removing this line breaks applying the language settings.
You can try it out if you change the language, save, close the game and open it. The new save won't be applied, but the drop down menu shows the selected language.

File: src/general/OptionsMenu.cs [hunk 2]
```diff
@@ -682,7 +682,9 @@ private void LoadLanguages(OptionButton optionButton)
 
         foreach (var locale in languages)
         {
-            optionButton.AddItem(locale);
+            var currentCulture = Settings.GetCultureInfo(locale);
+            var native = currentCulture.NativeName;
+            optionButton.AddItem(locale + "(" + native + ")");
```
Comment (FuJa0815) [inline src/general/OptionsMenu.cs:687]:
```suggestion
            optionButton.AddItem(locale + " (" + native + ")");
```

File: src/general/OptionsMenu.cs [hunk 3]
```diff
@@ -1058,7 +1060,15 @@ private void OnCustomUsernameTextChanged(string text)
...
-        Settings.Instance.SelectedLanguage.Value = languageSelection.GetItemText(item);
+        languages = TranslationServer.GetLoadedLocales().Cast<string>().OrderBy(i => i, StringComparer.InvariantCulture)
+            .ToList();
+        List<string> languageList = new List<string>();
+        foreach (var locale in languages)
+        {
+            languageList.Add(locale);
+        }
+
+        Settings.Instance.SelectedLanguage.Value = languageList[item];
```
Comment (FuJa0815) [inline src/general/OptionsMenu.cs:1071]: wohah, why did you do that so complicated?
languages already holds the loaded locales sorted, therefore you don't need line 1063-1064
in lines 1065-1069 you copy over the contents of `languages` into `languageList` and used `languageList`. Why didn't you just use the `languages` list directly?
```suggestion
        Settings.Instance.SelectedLanguage.Value = languages[item];
```

File: src/engine/Settings.cs (hunk 1 above)
Comment (FuJa0815) [reply src/engine/Settings.cs:352]: You can also see the error by changing the language and pressing the reset button.

**Review by FuJa0815 — CHANGES_REQUESTED** [review]

File: src/general/OptionsMenu.cs (hunk 3 above)
Comment (zackdrake) [reply src/general/OptionsMenu.cs]: I thought i needed to reset languages but i was wrong, sorry for making it more complicated than it should have been.
I made all the changes you suggested, thanks for your review

File: src/general/OptionsMenu.cs [hunk 4]
```diff
@@ -682,7 +682,9 @@ private void LoadLanguages(OptionButton optionButton)
 
         foreach (var locale in languages)
         {
-            optionButton.AddItem(locale);
+            var currentCulture = Settings.GetCultureInfo(locale);
+            var native = currentCulture.NativeName;
+            optionButton.AddItem(locale + "( " + native + ")");
```
Comment (FuJa0815) [inline src/general/OptionsMenu.cs:687]: That whitespace is placed wrong:
```suggestion
            optionButton.AddItem(locale + " (" + native + ")");
```

**Review by FuJa0815 — CHANGES_REQUESTED** [review]
Way better now

**Review by FuJa0815 — APPROVED** [review]

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**